mars/dataframe/base/get_dummies.py

```diff
@@ -0,0 +1,283 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import pandas as pd
+
+from ...core import OutputType, recursive_tile
+from ...serialization.serializables import StringField, BoolField, AnyField, ListField
+from ..core import SERIES_TYPE
+from ..datasource.dataframe import from_pandas as from_pandas_df
+from ..datasource.series import from_pandas as from_pandas_series
+from ..initializer import Series as asseries
+from ..operands import DataFrameOperand, DataFrameOperandMixin
+from ..reduction.unique import unique
+
+_encoding_dtype_kind = ['O', 'S', 'U']
+
+
+class DataFrameGetDummies(DataFrameOperand, DataFrameOperandMixin):
+    prefix = AnyField('prefix')
+    prefix_sep = StringField('prefix_sep')
+    dummy_na = BoolField('dummy_na')
+    columns = ListField('columns')
+    sparse = BoolField('sparse')
+    drop_first = BoolField('drop_first')
+    dtype = AnyField('dtype')
+
+    def __init__(self, prefix=None, prefix_sep="_", dummy_na=False,
+                 columns=None, sparse=False, drop_first=False, dtype=None, **kws):
+        super().__init__(prefix=prefix, prefix_sep=prefix_sep, dummy_na=dummy_na,
+                         columns=columns, sparse=sparse, drop_first=drop_first, dtype=dtype, **kws)
+        self.output_types = [OutputType.dataframe]
+
+    @classmethod
+    def tile(cls, op):
+        inp = op.inputs[0]
+        out = op.outputs[0]
+        if len(inp.chunks) == 1 or isinstance(inp, SERIES_TYPE):
+            chunk_op = op.copy().reset_key()
+            chunk_param = out.params
+            chunk_param['index'] = (0, 0)
+            chunk = chunk_op.new_chunk(inp.chunks, kws=[chunk_param])
+            new_op = op.copy().reset_key()
+            param = out.params
+            param['chunks'] = [chunk]
+            param['nsplits'] = ((np.nan,), (np.nan,))
+            return new_op.new_dataframe(op.inputs, kws=[param])
+        else:
+            if op.columns:
+                encoding_columns = op.columns
+            else:
+                encoding_columns = []
+                for idx, dtype in enumerate(inp.dtypes.values):
+                    if dtype.kind in _encoding_dtype_kind:
+                        column = inp.dtypes.index[idx]
+                        encoding_columns.append(column)
+            # reindex, make encoding columns in the end of dataframe, to keep pace with pandas.get_dummies
+            total_columns = list(inp.columns.to_pandas().array)
+            for col in encoding_columns:
+                total_columns.remove(col)
+            total_columns.extend(encoding_columns)
+            inp = yield from recursive_tile(inp[total_columns])
+
+            chunks = []
```

Comment:
The tile logic seems not efficient. I think we can split tile into two steps.

First, get unique chunks for each selected columns
``` Python
unique_chunks = dict()
for col in seleceted_columns:
    unique_chunks[col] = yield from recursive_tile(unique(inp[chunk_column]))
``` 
Second, select the needed unique chunks to create a new chunk:
``` Python
for chunk in inp.chunks:
    inputs = [chunk] + [unique_chunks[col] for col in chunk_columns]
    chunk_op.new_chunk(c, kws=[chunk_param])
```